tests/functional_small/type_traits/traits_shared.hpp

```diff
@@ -45,68 +45,88 @@
 // ************************************************************************
 //@HEADER
 */
+namespace pressio { namespace traits { namespace test {
+// -------------------------------------------------
 
 /*
-  Verify values of common container traits
+  Verify package predicates
 */
 
+template <typename T>
+void test_is_not_eigen_container() {
+#ifdef PRESSIO_ENABLE_TPL_EIGEN
+  static_assert(pressio::is_native_container_eigen<T>::value == false, "");
+  static_assert(pressio::is_vector_eigen<T>::value == false, "");
+  static_assert(pressio::is_dynamic_vector_eigen<T>::value == false, "");
+  static_assert(pressio::is_static_vector_eigen<T>::value == false, "");
+  static_assert(pressio::is_dynamic_row_vector_eigen<T>::value == false, "");
+  static_assert(pressio::is_static_row_vector_eigen<T>::value == false, "");
+  static_assert(pressio::is_dynamic_column_vector_eigen<T>::value == false, "");
+  static_assert(pressio::is_static_column_vector_eigen<T>::value == false, "");
+  static_assert(pressio::is_dense_matrix_eigen<T>::value == false, "");
+  static_assert(pressio::is_sparse_matrix_eigen<T>::value == false, "");
+  static_assert(pressio::is_static_dense_matrix_eigen<T>::value == false, "");
+  static_assert(pressio::is_dynamic_dense_matrix_eigen<T>::value == false, "");
+  static_assert(pressio::is_dense_row_major_matrix_eigen<T>::value == false, "");
+  static_assert(pressio::is_dense_col_major_matrix_eigen<T>::value == false, "");
+#endif
+}
+
+template <typename T>
+void test_is_not_teuchos_container() {
+#ifdef PRESSIO_ENABLE_TPL_TRILINOS
+  static_assert(pressio::is_dense_vector_teuchos<T>::value == false, "");
+  static_assert(pressio::is_dense_matrix_teuchos<T>::value == false, "");
+#endif
+}
+
+template <typename T>
+void test_is_not_epetra_container() {
+#ifdef PRESSIO_ENABLE_TPL_TRILINOS
+  static_assert(pressio::is_vector_epetra<T>::value == false, "");
+  static_assert(pressio::is_multi_vector_epetra<T>::value == false, "");
+#endif
+}
+
+template <typename T>
+void test_is_not_tpetra_container() {
```

Comment:
can you separate tpetra from tpetra_block?

Reply:
sure - fixed on dd029258dbe7be40c965dced2f9577a36e1dcaf4